### DISCUSSION TITLE
Could not find seed in imported module np.random ! Could not find RandomState in imported module np.random

### DISCUSSION THREAD

**Discussion by madou-sow** [pyccel/pyccel#1038] [Ideas]
hello I would like to convert these 2 files in python in C, I would like to know on the one hand if my project is playable. I send you the 2 files and the error messages that I have. I thank you for your collaboration

- Script 1 
```
import numpy as np

rng = np.random.RandomState(42)
#rng.randn(100, 2)

# Generate train data
X = 0.3 * rng.randn(100, 2)
X_train = np.r_[X + 2, X - 2]

# Generate some regular novel observations
X = 0.3 * rng.randn(20, 2)
X_test = np.r_[X + 2, X - 2]

# Generate some abnormal novel observations
X_outliers = rng.uniform(low=-4, high=4, size=(20, 2))

# Print the resultat
#print(np.random.rand())
#print(np.random.RandomState(42))
#print(np.random.rand())
#print(rng.randn(100, 2))
print("=== X ===")
print(X)
print("=== X_train ===")
print(X_train)
print("=== X ===")
print(X)
print("=== X_test ===")
print(X_test)
print("=== X_outliers ===")
print(X_outliers)

```

- Message Error 1 
```
pyccel test-random-randomstate.py --language c

ERROR at annotation (semantic) stage
pyccel:
 |fatal [semantic]: test-random-randomstate.py [3,6]| Could not find RandomState in imported module np.random (np.random.RandomState(FunctionCallArgument(Literal(42))))

```

- Script 2
```
import numpy as np

rng = np.random.seed(42)
#rng.randn(100, 2)

# Generate train data
X = 0.3 * np.random.rand(100, 2)
X_train = np.r_[X + 2, X - 2]

# Generate some regular novel observations
X = 0.3 * np.random.rand(20, 2)
X_test = np.r_[X + 2, X - 2]

# Generate some abnormal novel observations
X_outliers = np.random.uniform(low=-4, high=4, size=(20, 2))

# Print the resultat
#print(np.random.rand())
#print(np.random.RandomState(42))
#print(np.random.rand())
#print(rng.randn(100, 2))
print("=== X ===")
print(X)
print("=== X_train ===")
print(X_train)
print("=== X ===")
print(X)
print("=== X_test ===")
print(X_test)
print("=== X_outliers ===")
print(X_outliers)

```

- Message Error 2
```
pyccel test-random-seek.py --language c

ERROR at annotation (semantic) stage
pyccel:
 |fatal [semantic]: test-random-seek.py [3,6]| Could not find seed in imported module np.random (np.random.seed(FunctionCallArgument(Literal(42))))


```

**Comment by EmilyBourne** [comment]
Hi, please see the tutorial about [numpy](https://github.com/pyccel/pyccel/blob/master/tutorial/numpy-functions.md) for details of which functions are currently implemented. Currently, there are very few functions implemented from the `random` module (`rand` and `randint`) and these functions are only implemented for Fortran. We should be able to add some additional functionalities (e.g. `rand` for C or a seed) relatively easily, but we have not yet had a discussion about what the generated code should look like. If you have a clear idea of what the generated code should be then I would recommend opening an issue for each function that you would like to see with a clear description of what is required so that there is enough information for us to implement it.

Similarly numpy's [RClass](https://numpy.org/devdocs/reference/generated/numpy.r_.html) object is also not implemented currently. I think it would be nice to implement some functions like `column_stack`, `concatenate`, but `_r` seems like a very complicated use case. Especially if the string parameter is used.

For the moment it is possible to fill an array in a similar way using:
```
import numpy as np
X = np.random.rand(100, 2)
X *= 0.3
X_train = np.empty((200,2))
X_train[:100,:] = X+2
X_train[100:,:] = X-2
```
This code will translate to fortran but not c
